src/azure-cli/azure/cli/command_modules/containerapp/_clients.py

```diff
@@ -0,0 +1,1512 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long, consider-using-f-string, logging-format-interpolation, inconsistent-return-statements, broad-except, bare-except, too-many-statements, too-many-locals, too-many-boolean-expressions, too-many-branches, too-many-nested-blocks, pointless-statement, expression-not-assigned, unbalanced-tuple-unpacking, unsupported-assignment-operation
+# pylint: disable=super-with-arguments, too-many-instance-attributes, no-else-return, no-self-use, useless-return, reimported, redefined-outer-name, no-else-raise
+
+import json
+import time
+import sys
+
+from azure.cli.core.azclierror import AzureResponseError, ResourceNotFoundError
+from azure.cli.core.util import send_raw_request
+from azure.cli.core.commands.client_factory import get_subscription_id
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+CURRENT_API_VERSION = "2023-05-01"
+POLLING_TIMEOUT = 600  # how many seconds before exiting
+POLLING_SECONDS = 2  # how many seconds between requests
+POLLING_TIMEOUT_FOR_MANAGED_CERTIFICATE = 1500  # how many seconds before exiting
+POLLING_INTERVAL_FOR_MANAGED_CERTIFICATE = 4  # how many seconds between requests
+HEADER_AZURE_ASYNC_OPERATION = "azure-asyncoperation"
+HEADER_LOCATION = "location"
+
+
+class PollingAnimation():
+    def __init__(self):
+        self.tickers = ["/", "|", "\\", "-", "/", "|", "\\", "-"]
+        self.currTicker = 0
+
+    def tick(self):
+        sys.stderr.write('\r')
+        sys.stderr.write(self.tickers[self.currTicker] + " Running ..")
+        sys.stderr.flush()
+        self.currTicker += 1
+        self.currTicker = self.currTicker % len(self.tickers)
+
+    def flush(self):
+        sys.stderr.flush()
+        sys.stderr.write("\r\033[K")
+
+
+def poll(cmd, request_url, poll_if_status):  # pylint: disable=inconsistent-return-statements
```

Comment:
is this one still used?

Reply:
ContainerappsJobClient and GithubClients still use it.

Reply:
Changed to poll_status and poll_sesult